### DISCUSSION TITLE
Polymorphic responsible?

### DISCUSSION THREAD

**Discussion by kvictormatheusk** [palkan/logidze#191] [Ideas]
Hello!
I am migrating from Audit to Logidze and I have a question. Is there a way to, instead of just storing the ID, also store the Type of the person responsible for the change in a Logidze :: History?
I thought of solving this in the following ways:

` Logidze.with_meta({responsible_id: User.first.id, responsible_type: "User"})`
or
` Logidze.with_meta({"_r": {responsible_id: User.first.id, responsible_type: "User"}})`

But I would like to know if anyone has had this problem and if there is a better solution (even with monkey paths)

**Comment by palkan** [comment]
> I thought of solving this in the following ways:
Logidze.with_meta({responsible_id: User.first.id, responsible_type: "User"})
or
Logidze.with_meta({"_r": {responsible_id: User.first.id, responsible_type: "User"}})

I think, that's the right approach. Since we provide an ability to store custom metadata, it's up to the end user to design its structure.
